lib/src/main/java/com/auth0/jwt/oicmsg/KeyBundle.java

```diff
@@ -0,0 +1,519 @@
+                                key = new RSAKey("use");
+                                break;
+                            case "EC":
+                                key = new ECKey("use");
+                                break;
+                            case "SYMKey":
+                                key = new SYMKey("use");
+                                break;
+                            default:
+                                throw new IllegalArgumentException("Encryption type: " + typeIndex + " isn't supported");
+                        }
+                    } catch (JWKException exception) {
+                        logger.warn("While loading keys: " + exception);
+                        isSuccess = false;
+                    }
+
+                    if (isSuccess) {
+                        this.keys.add(key);
+                        flag = true;
+                        break;
+                    }
+                }
+            }
+
+            if (!flag) {
+                logger.warn("While loading keys, UnknownKeyType: " + kty);
+            }
+        }
+    }
+
+    private static List<String> harmonizeUsage(List<String> uses) {
+        Set<String> keys = map.keySet();
+        Set<String> usagesSet = new HashSet<>();
+        for (String use : uses) {
+            if (keys.contains(use)) {
+                usagesSet.add(use);
+            }
+        }
+        return new ArrayList<>(usagesSet);
+    }
+
+    public void doLocalJwk(String fileName) throws UpdateFailed {
+        JSONParser parser = new JSONParser();
+        try {
+            Object obj = parser.parse(new FileReader(
+                    fileName));
+            JSONObject jsonObject = (JSONObject) obj;
+            JSONArray keys = (JSONArray) jsonObject.get("keys");
+            Iterator<String> iterator = keys.iterator();
+            List<Key> keysList = new ArrayList<Key>();
+            while (iterator.hasNext()) {
+                keysList.add(new Gson().fromJson(iterator.next(), Key.class));
+            }
+            doKeys(keysList);
+        } catch (Exception e) {
+            logger.error("Now 'keys' keyword in JWKS");
+            throw new UpdateFailed("Local key updated from " + fileName + " failed.");
+        } finally {
+            this.lastUpdated = System.currentTimeMillis();
+        }
+    }
+
+    public void doLocalDer(String fileName, String keyType, List<String> keyUsage) throws NotImplementedException {
+        RSAKey rsaKey = rsaLoad(fileName);
+
+        if (!keyType.equalsIgnoreCase("rsa")) {
+            throw new NotImplementedException();
+        }
+
+        if (keyUsage.isEmpty()) {
+            keyUsage = new ArrayList<String>() {{
+                add("enc");
+                add("sig");
+            }};
+        } else {
+            keyUsage = harmonizeUsage(keyUsage);
+        }
+
+        for (String use : keyUsage) {
+            RSAKey key = new RSAKey().loadKey(rsaKey);
```

Comment:
This empty constructor in RSAKey doesn't exist. Also it's weird that you create a key and "load" another one. Maybe you need a different constructor or even a static method to create a new key from another one. What is it that you wanted to achieve here?

Reply:
That's how it's done in Roland's code.  For some reason when I go to the empty constructor in Roland's code, it takes me to full-blown one.  Interesting.

Reply:
RSAKey and raskey are not of the same type. 
One (rsakey) is whatever representation your cryptolibrary uses.
RSAKey is a wrapper to allow for easy moving between the JWK representation of the key and the cryptolibrary's representation.